ReactiveCocoa/AnyObject+Lifetime.swift

```diff
@@ -0,0 +1,117 @@
+import Foundation
+import ReactiveSwift
+
+public extension Lifetime {
+	/// Retrive the associated lifetime of given object.
+	/// The lifetime is ends when the object is deallocated.
+	///
+	/// - parameters:
+	///   - object: An object that is the owner of the lifetime to be returns.
+	///
+	/// - returns: A lifetime that ends when the given object is deallocated.
+	public static func of(_ object: AnyObject) -> Lifetime {
+		enum Keys {
+			/// Holds the `Lifetime` of the object.
+			static let isSwizzled = AssociationKey<Bool>(default: false)
+
+			/// Holds the `Lifetime` of the object.
+			static let lifetime = AssociationKey<Lifetime?>(default: nil)
+
+			/// Holds the `Lifetime.Token` of the object.
+			static let lifetimeToken = AssociationKey<Lifetime.Token?>(default: nil)
+		}
+
+		return synchronized(object) {
+			if let object = object as? NSObject {
+				if let lifetime = object.associations.value(forKey: Keys.lifetime) {
```

Comment:
I would rather prefer keeping `NSObject+Lifetime.swift` with the `NSObject` specific implementation and forward the call here to avoid this nesting TBH.